### DISCUSSION TITLE
Does this support Universal (Web) Apps?

### DISCUSSION THREAD

**Discussion by cyrfer** [maplibre/maplibre-react-native#503] [Q&A] [answered]
## Motivation
I just want to know if this supports web applications, as many React Native libraries do these days.

I am seeing an error with MapLibre JS and I'm looking for alternatives.

**Comment by sarahsporck** [comment, marked as answer]
Nope. This does only support native apps. Feel free to create an issue in the https://github.com/maplibre/maplibre-gl-js repository, if you have any issues.

**Reply by cyrfer** [reply]
Isn't the RN library responsible for being universal? How could adding an issue to JS library help if it needs to be in RN?

**Reply by KiwiKilian** [reply]
> Isn't the RN library responsible for being universal?

RN originated to support Android and iOS from a single codebase. Maintaining a RN library in general is a complex task. Supporting another platform is therefore raising this complexity even more. Right now the focus within this library is on Android and iOS.

From your initial question:

> I am seeing an error with MapLibre JS and I'm looking for alternatives.

That's why Sarah probably recommended opening an issue within the `maplibre-gl-js` repository. If this library would also support the Web platform, it would also leverage `maplibre-gl-js`. This library is basically an adapter, to unify the different Platform SDKs into one interface. It's not responsible itself to render a map.
